crates/interledger-service-util/src/statistics_service.rs

```diff
@@ -0,0 +1,89 @@
+//! An Interledger service that collects a variety of
+//! metrics about the packets going through it.
+
+use futures::Future;
+use interledger_service::{
+    Account, BoxedIlpFuture, IncomingRequest, IncomingService, OutgoingRequest, OutgoingService,
+};
+use metrics::{counter, timing};
+use std::{marker::PhantomData, time::Instant};
+
+#[derive(Clone)]
+pub struct IncomingStatisticsService<I, A> {
+    next: I,
+    account_type: PhantomData<A>,
+}
+
+impl<I, A> IncomingStatisticsService<I, A>
+where
+    I: IncomingService<A>,
+    A: Account,
+{
+    pub fn new(next: I) -> Self {
+        IncomingStatisticsService {
+            next,
+            account_type: PhantomData,
+        }
+    }
+}
+
+impl<I, A> IncomingService<A> for IncomingStatisticsService<I, A>
+where
+    I: IncomingService<A>,
+    A: Account,
+{
+    type Future = BoxedIlpFuture;
+
+    fn handle_request(&mut self, request: IncomingRequest<A>) -> Self::Future {
+        counter!("requests.incoming.prepare", 1);
+        let start_time = Instant::now();
+
+        Box::new(self.next.handle_request(request).then(move |result| {
+            if result.is_ok() {
+                counter!("requests.incoming.fulfill", 1);
```

Comment:
Maybe also add a "local" counter per account username? 


```suggestion
counter!("requests.incoming.fulfill", 1)
counter!(format!({}.requests.incoming.fulfill, request.from.username()", 1)
```